src/collective/solr/browser/maintenance.py

```diff
@@ -127,7 +127,12 @@ def reindex(self, batch=1000, skip=0, limit=0, ignore_portal_types=None,
         def checkPoint():
             for my_boost_values, data in updates.values():
                 adder = data.pop('_solr_adder')
-                adder(conn, boost_values=my_boost_values, **data)
+                try:
+                    adder(conn, boost_values=my_boost_values, **data)
+                except Exception, e:
```

Comment:
Hm, not 100% sure ...
From the ZODB I'd expect only read conflicts during reindexing and I'd say they should be handled like other exceptions. But maybe I'm missing something.